### PR TITLE
ci: standardize Github Actions

### PR DESCRIPTION
**Pull request by cpanato**
## Description
This Pr tries to make the CI jobs similar to the others in Org.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Related to https://github.com/openfaas/faas/issues/1643

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

/assign @alexellis

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ jobs:
         go-version: [1.15.x]
         os: [ubuntu-latest]
     runs-on: ${{ matrix.os }}
+
```
Comment (alexellis) [inline .github/workflows/publish.yaml:15]: This cannot be merged until we add the `permissions` section, please see: https://github.com/openfaas/ingress-operator/blob/master/.github/workflows/publish.yaml#L11

**Review by alexellis — CHANGES_REQUESTED** [review]
Quick change required.

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (alexellis) [reply .github/workflows/publish.yaml:15]: Only change the publish step though, as the build should not do any pushes of images, so doesn't need write access to packages.

Comment (cpanato) [reply .github/workflows/publish.yaml]: nice, this is a new thing, i was not aware 😄 

done, PTAL

**Pull request merged**